ignite/engine/engine.py

```diff
@@ -695,25 +695,32 @@ def _internal_run(self) -> State:
                     self._setup_engine()
 
                 time_taken = self._run_once_on_dataset()
+                # time is available for handlers but must be update after fire
                 self.state.times[Events.EPOCH_COMPLETED.name] = time_taken
+                if self.should_terminate:
```

Comment:
Maybe, it would be better to measure both cases: `self._fire_event(Events.TERMINATE)` and `self._fire_event(Events.EPOCH_COMPLETED)` ?